### ISSUE TITLE
Crate is not very portable due to usage of AtomicU64

### ISSUE THREAD

**Issue by arthurprs** [robinfriedli/rusty_pool#3]
Hello :wave:  right now compiling this crate fail on some targets due to the lack of AtomicU64.

I was thinking if the worker could maybe be an AtomicUSize instead?
The downside is that 32bit targets would be limited 64K workers, which should be fine IMO.

What do you think?

**Comment by robinfriedli** [comment]
Good point! Currently, bookkeeping depends on it being 64 bits as the total worker counter and idle worker counter are two 32 bit values in an AtomicU64, the total worker count is read by shifting the counter to the right by 32 bits, the idle worker count is read by masking the bits with `0x0000_0000_FFFF_FFFF` and the total worker count is incremented by adding `0x0000_0001_0000_0000` and so on. However, this should be a fairly simple change and since `std::mem::size_of` is a const function, it should be doable to use the size of usize on the given platform for those arithmetics dynamically without adding runtime overhead. For example `const WORKER_IDLE_MASK: u64 = 0x0000_0000_FFFF_FFFF;` could be replaced with
```rust
const BITS: usize = std::mem::size_of::<usize>();
const WORKER_IDLE_MASK: usize = (1 << (BITS / 2)) - 1;
```

> The downside is that 32bit targets would be limited 64K workers, which should be fine IMO.

Certainly beats not being able to compile the crate at all  😅

**Comment by robinfriedli** [comment]
I opened a pull request if you want to check it out. Since changing function signatures of the public API is a breaking change the change will be released as 0.6.0.

**Comment by robinfriedli** [comment]
Change has been released with 0.6.0

**Comment by arthurprs** [comment]
Thank you :tada:

Solved in #4
